spec/integration/nested_attribute_spec.rb

```diff
@@ -52,80 +52,98 @@ class CarWithLimit < Car
+    expect(Bar.find(bar1.id).uno).to eq 'bar1 uno'
+    expect(Bar.where(id: bar2.id).first).to be_nil
+    expect(Bar.where(uno: "newbar uno").first).to_not be_nil
+
+    bars = car.bars(true)
+    expect(bars).to include(bar1)
+    expect(bars).to_not include(bar2)
   end
 
-  it "should reject attributes when all blank" do
-    @car, @bar1, @bar2 = create_car_with_bars(CarAllBlank)
-
-    expect(@car.bars.count).to eq 2
-    @car.update bars_attributes: [{}, {:id=>@bar1.id, :uno=>"bar1 uno"}]
-    expect(@car.bars(true).count).to eq 2
+  describe "reject_if: :all_blank" do
+    let(:car_class) { CarAllBlank }
+    it "should reject attributes when all blank" do
+      expect(car.bars.count).to eq 2
+      car.update bars_attributes: [{}, {id: bar1.id, uno: "bar1 uno"}]
+      expect(car.bars(true).count).to eq 2
 
-    @bar1.reload
-    expect(@bar1.uno).to eq "bar1 uno"
+      bar1.reload
+      expect(bar1.uno).to eq "bar1 uno"
+    end
   end
 
-  it "should reject attributes based on proc" do
-    @car, @bar1, @bar2 = create_car_with_bars(CarProc)
-
-    @car.update bars_attributes: [{}, {:id=>@bar1.id, :uno=>"bar1 uno"}, {:id=>@bar2.id, :dos=>"bar2 dos"}]
-    @bar1.reload
-    @bar2.reload
-    expect(@bar1.uno).to eq "bar1 uno"
-    expect(@bar2.dos).to be_nil
+  describe "reject_if: with a proc" do
+    let(:car_class) { CarProc }
+    it "should reject attributes based on proc" do
+      car.update bars_attributes: [{}, {id: bar1.id, uno: "bar1 uno"}, {id: bar2.id, dos: "bar2 dos"}]
+      bar1.reload
+      bar2.reload
+      expect(bar1.uno).to eq "bar1 uno"
+      expect(bar2.dos).to be_nil
+    end
   end
 
-  it "should reject attributes base on method name" do
-    @car, @bar1, @bar2 = create_car_with_bars(CarSymbol)
+  describe "allow_destroy: false" do
+    let(:car_class) { CarProc }
+    it "should ignore _destroy flag and reject based on proc" do
+      car.update bars_attributes: [{id: bar2.id, dos: "bar2 dos", _destroy:"1"}]
+      bar2.reload
+      expect(bar2.dos).to be_nil
+    end
 
-    @car.update bars_attributes: [{}, {:id=>@bar1.id, :uno=>"bar1 uno"}, {:id=>@bar2.id, :dos=>"bar2 dos"}]
-    @bar1.reload
-    @bar2.reload
-    expect(@bar1.uno).to eq "bar1 uno"
-    expect(@bar2.dos).to be_nil
+    it "should create a new record even if _destroy is set" do
+      expect(car.bars.count).to eq 2
+      car.update bars_attributes: [{uno: "new uno", _destroy: "1"}]
+      expect(car.bars(true).count).to eq 3
+    end
   end
 
-  it "should throw TooManyRecords" do
-    @car, @bar1, @bar2 = create_car_with_bars(CarWithLimit)
-
-    expect {
-      @car.attributes = {:bars_attributes=>[{}]}
-    }.to_not raise_exception
-
-    expect {
-      @car.attributes = {:bars_attributes=>[{}, {}]}
-    }.to raise_exception(ActiveFedora::NestedAttributes::TooManyRecords)
+  describe "allow_destroy: true" do
+    let(:car_class) { Car }
+    it "should not create a new record if _destroy is set" do
```

Comment:
I disagree with this. If you set _destroy and you don't set an id, it should ignore the _destroy flag and go on creating as usual.